### ISSUE TITLE
Forms 1.1.1 is completely broken

### ISSUE THREAD

**Issue by pzehle** [nextcloud/forms#229]
### Steps to reproduce
1. Install forms app (if it is installed, delete it and re-install).
2. As an admin, click on forms, then new form, and reload page.
3. As a regular user click on forms.

### Expected behaviour
To be able to create a form, edit it, add answers, share, as an admin and as a normal user.

### Actual behaviour
As an admin, after clicking on create form, and reloading the page, it says _Access forbidden, CSRF check failed_. No forms can be created.
As a normal user, clicking on forms result on a redirection to / and a message in the console:

`TypeError: Cannot read property 'call' of undefined
    at o (forms.js?v=30498835-10:1)
    at Object.<anonymous> (1.js:68)
    at n (1.js:1)
    at Module.<anonymous> (1.js:68)
    at n (1.js:1)
    at 1.js:1
    at Object.<anonymous> (1.js:1)
    at o (forms.js?v=30498835-10:1)
    at Object.227 (5.js:1)
    at o (forms.js?v=30498835-10:1)`

## Server configuration detail

**Operating system:** Linux 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64

**Webserver:** Apache/2.4.29 (Ubuntu) (apache2handler)

**Database:** mysql 10.4.12

**PHP version:** 

7.2.24-0ubuntu0.18.04.3
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, sodium, session, standard, apache2handler, mysqlnd, PDO, xml, apcu, apc, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, imagick, intl, json, exif, mysqli, pdo_mysql, Phar, posix, readline, shmop, SimpleXML, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, Zend OPcache

**Nextcloud version:** 18.0.1 - 18.0.1.3

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** unknown

<details><summary>Signing status</summary>

Array
(
)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.4.0
 - admin_audit: 1.8.0
 - announcementcenter: 3.7.0
 - bruteforcesettings: 1.5.0
 - cloud_federation_api: 1.1.0
 - comments: 1.8.0
 - dav: 1.14.0
 - drawio: 0.9.5
 - extract: 1.2.3
 - federatedfilesharing: 1.8.0
 - federation: 1.8.0
 - files: 1.13.1
 - files_accesscontrol: 1.8.1
 - files_mindmap: 0.0.18
 - files_pdfviewer: 1.7.0
 - files_rightclick: 0.15.2
 - files_sharing: 1.10.1
 - files_trashbin: 1.8.0
 - files_versions: 1.11.0
 - files_videoplayer: 1.7.0
 - firstrunwizard: 2.7.0
 - forms: 1.1.1
 - guests: 1.4.1
 - issuetemplate: 0.6.0
 - logreader: 2.3.0
 - lookup_server_connector: 1.6.0
 - nextcloud_announcements: 1.7.0
 - notifications: 2.6.0
 - oauth2: 1.6.0
 - password_policy: 1.8.0
 - passwords: 2020.3.1
 - privacy: 1.2.0
 - provisioning_api: 1.8.0
 - recommendations: 0.6.0
 - serverinfo: 1.8.0
 - settings: 1.0.0
 - sharebymail: 1.8.0
 - spreed: 8.0.5
 - support: 1.1.0
 - survey_client: 1.6.0
 - systemtags: 1.8.0
 - text: 2.0.0
 - theming: 1.9.0
 - twofactor_backupcodes: 1.7.0
 - updatenotification: 1.8.0
 - viewer: 1.2.0
 - workflowengine: 2.0.0
Disabled:
 - activity
 - calendar
 - contacts
 - cospend
 - deck
 - encryption
 - external
 - files_external
 - mail
 - notes
 - onlyoffice
 - photos
 - polls
 - tasks
 - timetracker
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "cloud.artn.digital"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "18.0.1.3",
    "overwrite.cli.url": "https:\/\/cloud.artn.digital",
    "htaccess.RewriteBase": "\/",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "mail_smtpmode": "smtp",
    "mail_smtpsecure": "tls",
    "mail_sendmailmode": "smtp",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpauth": 1,
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "587",
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpauthtype": "LOGIN",
    "memcache.local": "\\OC\\Memcache\\APCu",
    "maintenance": false
}
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** 

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

## Client configuration

**Browser:** Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.132 Safari/537.36

**Operating system:** 

## Logs

<details><summary>Web server error log</summary>

```

```
</details>

<details><summary>Nextcloud log</summary>

```

```
</details>

<details><summary>Browser log</summary>
`TypeError: Cannot read property 'call' of undefined
    at o (forms.js?v=ca636f73-10:1)
    at Object.<anonymous> (1.js:68)
    at n (1.js:1)
    at Module.<anonymous> (1.js:68)
    at n (1.js:1)
    at 1.js:1
    at Object.<anonymous> (1.js:1)
    at o (forms.js?v=ca636f73-10:1)
    at Object.227 (5.js:1)
    at o (forms.js?v=ca636f73-10:1)`
</details>

**Comment by rogercreagh** [comment]
1. I can confirm that as an admin once you have created a form you can no longer edit it.

2, Also if as admin you respond to your own form you can no longer access it

3. Also the "only allow one submission per user" checkbox when creating a form is greyed out. (which presumably explains point 3)

I am unclear as to whether it is meant to be possible to edit a form after creating it (before any submissions are made)

**Comment by jotoeri** [comment]
> * I can confirm that as an admin once you have created a form you can no longer edit it.

That's true, editing forms is not possible yet, it's a feature to develop.

> * Also if as admin you respond to your own form you can no longer access it

This i could not reproduce yet?!

> * Also the "only allow one submission per user" checkbox when creating a form is greyed out.

This checkbox is only available when using registered-users access, as such it is possible to control which user submitted already.

**Comment by jotoeri** [comment]
@pzehle Do you still observe the Problem? I got some similar Problems during development, when my Browser still used old, cached js-Files. Clearing the cache resolved it.
Apart from that, it is somehow hard to reproduce or understand the error on my installation... :/

**Comment by pzehle** [comment]
@jotoeri I can confirm that deleting cache solves my problem. Although this is kind of unexpected, I am able to use the app now. Thank you.

**Comment by rogercreagh** [comment]
@jotoeri many thanks for the clarifications.

One the third point I tried selecting the Only Shared option and the One Submission remains grey'd. I must have not tried Reg Only. Since Only Shared is in effect a subset of registered users that you select by group or individuals, the only one submission per user option should be available there too., shouldn't it? (that's what I assumed :-) )

The second point I now can't reproduce - but looking at the form that generated the problem I suspect I may have tried to access it just after it had expired.
The message could be improved for an expired form - at present you get rather stentorian message saying 
**Access denied
You are not allowed to view this form or the form does not exist.**

In the case of an out of date form it would be more useful to say:
"Sorry, entries for this form closed on {{nicely formatted datetime}}"

My expectation that I would be able to edit my own form will hopefully arrive in due course - changing the close datetime (either to extend or close early) would probably be the most common use case. Changing questions once a first submission has been made is probably not a good idea (although before the form is opened previewing and being able to return to edit typos, change text, change order of questions and answer options would be very useful)

The documentation seems a bit sparse at present - would be happy to help improve that. If we decide to deploy it in our small voluntary organisation I'll probably write a user guide qhich I will share back here anyway.

**Comment by jotoeri** [comment]
@rogercreagh 
- For the only-shared option you are absolutely right, that sounds like a bug, but shouldn't be too hard to solve. Can you open a separate issue for that?
- For the expired-response: Same here. Your suggestion sounds useful, would you like to open a separate issue for that? ;)

- I think editing the form will come in any ways. We will see how, at the moment i would await for an already ongoing layout/code-cleanup, which seems to implement at least a clone-functionality already (see #237). We'll see on the next days/next week, i think. :)

- If you want to improve the documentation, would be surely appreciated. :) I would say, just wait for the large ongoing change to be finished... ;)

Greets,
Jonas

**Comment by jotoeri** [comment]
> @jotoeri I can confirm that deleting cache solves my problem. Although this is kind of unexpected, I am able to use the app now. Thank you.

@skjnldsv @rullzer Is this a common/accepted thing on updating apps, that errors occur if the Browser does not refresh all cached files? Or is there somewhere a code-snippet to implement, e.g. forcing the browsers to reload after update?
(I tend to say #228 could be a similar thing...)

**Comment by rullzer** [comment]
@jotoeri it actually should reload them. As the assets (if inserted the right way) have a hash that is updated.
I'll check if that is the case everywhere.

**Comment by skjnldsv** [comment]
> @skjnldsv @rullzer Is this a common/accepted thing on updating apps, that errors occur if the Browser does not refresh all cached files? Or is there somewhere a code-snippet to implement, e.g. forcing the browsers to reload after update?

yeah, it might be complicated to manage.
I'll check the config see if we can force the refresh with webpack :)

**Comment by jotoeri** [comment]
@skjnldsv @rullzer Any news on this? Issue solved?

**Comment by skjnldsv** [comment]
Did not checked yet

**Comment by LU1ZINH0** [comment]
Hello!

Have same issue, after upgrade 1.0.8 to 1.1.1... Same error, blank page. Never used form in 1.0.8

**Comment by skjnldsv** [comment]
Hello!
I will need your browser console log to investigate this issue.
Open your console, reload your page and/or do the action leading to this issue and copy/paste the log in this thread.

<details>
<summary>How to access your browser console (Click to expand)</summary>

# Chrome
- Press either CTRL + SHIFT + J to open the “console” tab of the Developer Tools.
- Alternative method:
    1. Press either CTRL + SHIFT + I or F12 to open the Developer Tools.
    2. Click the “console” tab.

# Safari
- Press CMD + ALT + I to open the Web Inspector.
- See Chrome’s step 2. (Chrome and Safari have pretty much identical dev tools.)

# IE9
1. Press F12 to open the developer tools.
2. Click the “console” tab.

# Firefox
- Press CTRL + SHIFT + K to open the Web console (COMMAND + SHIFT + K on Macs).
- or, if Firebug is installed (recommended):
    1. Press F12 to open Firebug.
    2. Click on the “console” tab.

# Opera
1. Press CTRL + SHIFT + I to open Dragonfly.
2. Click on the “console” tab.
</details>

**Comment by LU1ZINH0** [comment]
Hello.

I made a clone of the machine to apply NextCloud updates from version 18.0.0 to 18.0.3 and took the opportunity to understand what happens with Forms.

Just updating the NC to the latest version didn't solve it, so what I did:
1. I downloaded the version of Forms 1.0.4 (h ** ps: //github.com/affan98/forms/releases/download/v.1.0.4/forms.tar.gz), applied the permissions, owner the file and enabled forms via the OCC command line. FORMS in this version 1.0.4, it is "almost" functional, it is possible to create forms but they cannot be VIEWED ... although there is data in the database.

2. The first unsuccessful test was to update FORMS 1.0.4 to 1.1.1 via "occ app: update forms", the result of the console:

jquery-migrate.min.js:2 JQMIGRATE: Migrate is installed, version 1.4.1
ProxyBus.js:65 Proxying an event bus of version 1.1.2 with 0.2.1
t @ ProxyBus.js:65
(anonymous) @ index.js:20
n @ bootstrap:19
(anonymous) @ requesttoken.js:11
n @ bootstrap:19
(anonymous) @ index.js:25
n @ bootstrap:19
(anonymous) @ client.js:12
n @ bootstrap:19
(anonymous) @ notifications.js?v=367643a9-23:21
n @ bootstrap:19
(anonymous) @ bootstrap:83
(anonymous) @ notifications.js?v=367643a9-23:1
globals.js:66 Handlebars is deprecated: please ship your own, this will be removed in Nextcloud 20
ne @ globals.js:66
get @ globals.js:96
e.default @ no-conflict.js:9
(anonymous) @ handlebars.runtime.js:60
n @ bootstrap:19
(anonymous) @ runtime.js:3
n @ bootstrap:19
(anonymous) @ file.handlebars:1
n @ bootstrap:19
(anonymous) @ richObjectStringParser.js:13
n @ bootstrap:19
(anonymous) @ bootstrap:83
(anonymous) @ notifications.js?v=367643a9-23:1
forms.js?v=367643a9-23:67 TypeError: Cannot read property 'call' of undefined
    at o (forms.js?v=367643a9-23:1)
    at Object.<anonymous> (1.js:68)
    at n (1.js:1)
    at Module.<anonymous> (1.js:68)
    at n (1.js:1)
    at 1.js:1
    at Object.<anonymous> (1.js:1)
    at o (forms.js?v=367643a9-23:1)
    at Object.227 (5.js:1)
    at o (forms.js?v=367643a9-23:1)
i @ forms.js?v=367643a9-23:67
(anonymous) @ forms.js?v=367643a9-23:67
(anonymous) @ forms.js?v=367643a9-23:67
(anonymous) @ forms.js?v=367643a9-23:67
Promise.then (async)
(anonymous) @ forms.js?v=367643a9-23:67
(anonymous) @ forms.js?v=367643a9-23:67
(anonymous) @ forms.js?v=367643a9-23:67
Ot @ forms.js?v=367643a9-23:67
(anonymous) @ forms.js?v=367643a9-23:67
d @ forms.js?v=367643a9-23:67
r @ forms.js?v=367643a9-23:67
r @ forms.js?v=367643a9-23:67
_t @ forms.js?v=367643a9-23:67
Tt.confirmTransition @ forms.js?v=367643a9-23:67
Tt.transitionTo @ forms.js?v=367643a9-23:67
Ft.init @ forms.js?v=367643a9-23:67
beforeCreate @ forms.js?v=367643a9-23:67
Ut @ forms.js?v=367643a9-23:7
Ze @ forms.js?v=367643a9-23:7
t._init @ forms.js?v=367643a9-23:7
xn @ forms.js?v=367643a9-23:7
(anonymous) @ forms.js?v=367643a9-23:112
o @ forms.js?v=367643a9-23:1
(anonymous) @ forms.js?v=367643a9-23:1
(anonymous) @ forms.js?v=367643a9-23:1


3. The second test was to download directly (h ** ps: //github.com/nextcloud/forms/releases/download/v1.1.1/forms.tar.gz), the same process of defining permissions, owner and enable the form via occ. There was no change by the console, below:
jquery-migrate.min.js:2 JQMIGRATE: Migrate is installed, version 1.4.1
ProxyBus.js:65 Proxying an event bus of version 1.1.2 with 0.2.1
t @ ProxyBus.js:65
(anonymous) @ index.js:20
n @ bootstrap:19
(anonymous) @ requesttoken.js:11
n @ bootstrap:19
(anonymous) @ index.js:25
n @ bootstrap:19
(anonymous) @ client.js:12
n @ bootstrap:19
(anonymous) @ notifications.js?v=367643a9-23:21
n @ bootstrap:19
(anonymous) @ bootstrap:83
(anonymous) @ notifications.js?v=367643a9-23:1
globals.js:66 Handlebars is deprecated: please ship your own, this will be removed in Nextcloud 20
ne @ globals.js:66
get @ globals.js:96
e.default @ no-conflict.js:9
(anonymous) @ handlebars.runtime.js:60
n @ bootstrap:19
(anonymous) @ runtime.js:3
n @ bootstrap:19
(anonymous) @ file.handlebars:1
n @ bootstrap:19
(anonymous) @ richObjectStringParser.js:13
n @ bootstrap:19
(anonymous) @ bootstrap:83
(anonymous) @ notifications.js?v=367643a9-23:1
forms.js?v=367643a9-23:67 TypeError: Cannot read property 'call' of undefined
    at o (forms.js?v=367643a9-23:1)
    at Object.<anonymous> (1.js:68)
    at n (1.js:1)
    at Module.<anonymous> (1.js:68)
    at n (1.js:1)
    at 1.js:1
    at Object.<anonymous> (1.js:1)
    at o (forms.js?v=367643a9-23:1)
    at Object.227 (5.js:1)
    at o (forms.js?v=367643a9-23:1)
i @ forms.js?v=367643a9-23:67
(anonymous) @ forms.js?v=367643a9-23:67
(anonymous) @ forms.js?v=367643a9-23:67
(anonymous) @ forms.js?v=367643a9-23:67
Promise.then (async)
(anonymous) @ forms.js?v=367643a9-23:67
(anonymous) @ forms.js?v=367643a9-23:67
(anonymous) @ forms.js?v=367643a9-23:67
Ot @ forms.js?v=367643a9-23:67
(anonymous) @ forms.js?v=367643a9-23:67
d @ forms.js?v=367643a9-23:67
r @ forms.js?v=367643a9-23:67
r @ forms.js?v=367643a9-23:67
_t @ forms.js?v=367643a9-23:67
Tt.confirmTransition @ forms.js?v=367643a9-23:67
Tt.transitionTo @ forms.js?v=367643a9-23:67
Ft.init @ forms.js?v=367643a9-23:67
beforeCreate @ forms.js?v=367643a9-23:67
Ut @ forms.js?v=367643a9-23:7
Ze @ forms.js?v=367643a9-23:7
t._init @ forms.js?v=367643a9-23:7
xn @ forms.js?v=367643a9-23:7
(anonymous) @ forms.js?v=367643a9-23:112
o @ forms.js?v=367643a9-23:1
(anonymous) @ forms.js?v=367643a9-23:1
(anonymous) @ forms.js?v=367643a9-23:1

4. I also tested installing and enabling FORMS via Nextcloud's APPS, without success. The console log follows:
jquery-migrate.min.js:2 JQMIGRATE: Migrate is installed, version 1.4.1
ProxyBus.js:65 Proxying an event bus of version 1.1.2 with 0.2.1
t @ ProxyBus.js:65
globals.js:66 Handlebars is deprecated: please ship your own, this will be removed in Nextcloud 20
ne @ globals.js:66
forms.js?v=367643a9-23:67 TypeError: Cannot read property 'call' of undefined
    at o (forms.js?v=367643a9-23:1)
    at Object.<anonymous> (1.js:68)
    at n (1.js:1)
    at Module.<anonymous> (1.js:68)
    at n (1.js:1)
    at 1.js:1
    at Object.<anonymous> (1.js:1)
    at o (forms.js?v=367643a9-23:1)
    at Object.227 (5.js:1)
    at o (forms.js?v=367643a9-23:1)
i @ forms.js?v=367643a9-23:67

5. The curious thing is that if you use the "forms" link (https: //*****/index.php/apps/forms/) the page is blank but if I type (https: // ***** /index.php/apps/forms/new) I get the following error: Access forbidden - CSRF check failed


7. I don't know if it is relevant, but I ran it via the occ app command line: check-code, here is the result:

[root@ip-******** nextcloud]# sudo -u apache php occ app:check-code forms
Analysing /var/www/html/nextcloud/apps/forms/lib/Controller/ApiController.php
 1 errors
    line  217: == - is discouraged
Analysing /var/www/html/nextcloud/apps/forms/lib/Controller/PageController.php
 6 errors
    line   60: OCP\User - deprecated class must not be imported with a use statement
    line  300: User::checkLoggedIn - Method of deprecated class must not be called
    line  374: == - is discouraged
    line  455: Util::writeLog - Method of deprecated class must not be called
    line  532: Util::writeLog - Method of deprecated class must not be called
    line  567: Util::writeLog - Method of deprecated class must not be called
Analysing /var/www/html/nextcloud/apps/forms/lib/Db/Event.php
 1 errors
    line   91: OC_User - Static method of private class must not be called
Analysing /var/www/html/nextcloud/apps/forms/templates/vote.tmpl.php
 1 errors
    line   24: OCP\User - deprecated class must not be imported with a use statement
App is not compliant

Thanks in advance for your attention and help!

**Comment by skjnldsv** [comment]
Fixed with upcomming 2.0

**Comment by LU1ZINH0** [comment]
Thanks!.... Installed Beta Version manually, resolved issue! :) :) :)
